### ISSUE TITLE
DeepCopyFrom fails with Metadata as Attribute

### ISSUE THREAD

**Issue by jviau** [Microsoft/msbuild#1319]
DeepCopyFrom fails when it attempts to copy an element with metadata as attribute. This leads to an exception thrown and a partially copied project.

```
"The element <A> beneath element <Compile> is invalid because a child element with that name already exists"
```

``` Csharp
string testProject =
  @"<?xml version=""1.0"" encoding=""utf-8""?>
  <Project xmlns = ""http://schemas.microsoft.com/developer/msbuild/2003"">
    <ItemGroup>
      <Compile Include=""Class1.cs"" A=""a"" />
      <Compile Include=""Class2.cs"" />
    </ItemGroup>
  </Project>";

  var attributeProject = ProjectRootElement.Create(XmlReader.Create(new StringReader(testProject)));
  var elementProject = ProjectRootElement.Create();
  try
  {
    elementProject.RemoveAllChildren();
    elementProject.DeepCopyFrom(attributeProject);
  }
  catch (Exception e) // InvalidProjectException thrown
  {
  }

  Assert.AreEqual(elementProject.Items.Count, attributeProject.Items.Count); // Fails! "Class2.cs" is missing.
```

**Comment by cdmihai** [comment]
This leads to VS failing anytime the project has an attribute metadata and the project is edited on disk.

**Comment by jviau** [comment]
@cdmihai that scenario only occurs in the new DotNet core projects, it is the only one to perform a live reload off disc.

**Comment by cdmihai** [comment]
Fixed in #1323
